web/packages/teleport/src/DesktopSession/useTdpClientCanvas.tsx

```diff
@@ -140,14 +131,16 @@ export default function useTdpClientCanvas(props: Props) {
     });
   };
 
-  // TODO(zmb3): this is not what an info-level alert should do.
-  // rename it to something like onGracefulDisconnect
   const clientOnTdpInfo = (info: string) => {
-    setDirectorySharingState(defaultDirectorySharingState);
-    setClipboardSharingState(defaultClipboardSharingState);
-    setTdpConnection({
-      status: '', // gracefully disconnecting
-      statusText: info,
+    setAlerts(prevState => {
+      return [
+        ...prevState,
+        {
+          content: info,
+          severity: 'info',
```

Comment:
`TDP_INFO` is no longer used to signal disconnection, so now I show info warnings when we receive it.
But I'm not sure if it makes any sense, we don't have any code that sends info level alerts?

Reply:
The original intent was that the TDP "notification" message can be used for both fatal errors or non-fatal messages.

- `SeverityError` means there was a fatal error and the proxy is about to close the connection.
- `SeverityInfo` and `SeverityWarning` are for non-fatal messages that show up in the top right corner.

In reality, Isaiah used `SeverityInfo` to show connection close "errors" in a state that's not error, which is why there's nothing else that's using INFO level.

Maybe we need to treat these as two separate concepts:
- a severity, which indicates to the UI how to render the message
- whether or not this message is preceding a disconnect